step-04/index.js

```diff
@@ -29,15 +29,15 @@ io.sockets.on('connection', function(socket) {
   socket.on('create or join', function(room) {
     log('Received request to create or join room ' + room);
 
-    var numClients = io.sockets.sockets.length;
+    var clientsInRoom = io.sockets.adapter.rooms[room];
```

Comment:
But in that case you are logging before someone joins or creates a room right? The output that I showed was after users join or create a room.